bazarr/custom_lang.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import os
+
+from subzero.language import Language
+
+logger = logging.getLogger(__name__)
+
+
+class CustomLanguage:
+    """Base class for custom languages."""
+
+    alpha2 = "pb"
+    alpha3 = "pob"
+    language = "pt-BR"
+    official_alpha2 = "pt"
+    official_alpha3 = "por"
+    name = "Brazilian Portuguese"
+    iso = "BR"
+    _possible_matches = ("pt-br", "pob", "pb", "brazilian", "brasil", "brazil")
+    _extensions = (".pt-br", ".pob", ".pb")
+    _extensions_forced = (".pt-br.forced", ".pob.forced", ".pb.forced")
+
+    def subzero_language(self):
+        return Language(self.official_alpha3, self.iso)
+
+    @classmethod
+    def from_value(cls, value, attr="alpha3"):
+        """Return a custom language subclass by value and attribute
+        if found, otherwise return None.
+
+        :param value:
+        :param attr:
+        """
+        for sub in cls.__subclasses__():
+            if getattr(sub, attr) == str(value):
+                return sub()
+
+        return None
+
+    @classmethod
+    def register(cls, table):
+        " Register the custom language subclasses in the database. "
+
+        for sub in cls.__subclasses__():
+            table.insert(
+                {table.code3: sub.alpha3, table.code2: sub.alpha2, table.name: sub.name}
+            ).on_conflict(action="IGNORE").execute()
+
+    @classmethod
+    def found_external(cls, subtitle, subtitle_path):
+        for sub in cls.__subclasses__():
+            code = sub.get_alpha_type(subtitle, subtitle_path)
+            if code:
+                return code
+
+        return None
+
+    @classmethod
+    def get_alpha_type(cls, subtitle: str, subtitle_path=None):
+        assert subtitle_path is not None
+
+        extension = str(os.path.splitext(subtitle)[0]).lower()
+        to_return = None
+
+        if extension.endswith(cls._extensions):
+            to_return = cls.alpha2
+
+        if extension.endswith(cls._extensions_forced):
+            to_return = f"{cls.alpha2}:forced"
+
+        if to_return is not None:
+            logging.debug("BAZARR external subtitles detected: %s", to_return)
+
+        return to_return
+
+    def ffprobe_found(self, detected_language: dict) -> bool:
+        name = detected_language.get("name", "").lower()
+        if not name:
+            return False
+
+        return any(ext in name for ext in self._possible_matches)
+
+
+class BrazilianPortuguese(CustomLanguage):
+    pass
```

Comment:
Why a `pass` here? Because it's already implemented in subzero language? Maybe a comment to explain it would help to support it later on. :-)

Reply:
Because all the attributes are the same as the base class (`CustomLanguage`). There's no need to overwrite anything.